src/array/cpu/rowwise_pick.h

```diff
@@ -208,20 +209,21 @@ COOMatrix CSRRowWisePick(CSRMatrix mat, IdArray rows,
 
 // Template for picking non-zero values row-wise. The implementation utilizes
 // OpenMP parallelization on rows because each row performs computation independently.
-template <typename IdxType>
-COOMatrix CSRRowWisePerEtypePick(CSRMatrix mat, IdArray rows, IdArray etypes,
+template <typename IdxType, typename DType>
+COOMatrix CSRRowWisePerEtypePick(CSRMatrix mat, IdArray rows,
+                                 const std::vector<int64_t>& etype_offset,
                                  const std::vector<int64_t>& num_picks, bool replace,
-                                 bool etype_sorted, RangePickFn<IdxType> pick_fn) {
+                                 bool etype_sorted, RangePickFn<IdxType> pick_fn,
```

Comment:
Will `etype_sorted == false` but etype_offset is provided? It seems to be conflicting.

Reply:
`etype_sorted` refers to whether the CSR indices per row are ordered by edge type.  It's different from ordering the CSR edge IDs by edge type.  Renamed it as suggested.